bundles/org.eclipse.core.resources/src/org/eclipse/core/internal/resources/Workspace.java

```diff
@@ -1869,6 +1869,11 @@ private void initializePreferenceLookupOrder() {
 		newOrder.add(ProjectScope.SCOPE);
 		newOrder.addAll(Arrays.asList(original));
 		service.setDefaultDefaultLookupOrder(newOrder.toArray(new String[newOrder.size()]));
+		Preferences node = service.getRootNode().node(ProjectScope.SCOPE);
```

Comment:
I wonder if we should hard cast the node we get to `ProjectPreferences` here, anything else is most likely an error. But I see that no one else does that, so at least we should have a test that verifies the right type returned (in case someone refactors the code, this patch will silently break workspace). Ideally also an error logged in "else" branch.

For test, something in org.eclipse.core.tests.internal.resources.ProjectPreferencesTest.

But looking on this in debugger, the code here isn't really used, because `ProjectPreferences.getWorkspace()` internally uses `ResourcesPlugin.getWorkspace()`.

Is this intentional?

Reply:
> I wonder if we should hard cast the node we get to `ProjectPreferences` here

I find casts always a bit ugly but if anything else is an error might be suitable here.

> But looking on this in debugger, the code here isn't really used, because `ProjectPreferences.getWorkspace()` internally uses `ResourcesPlugin.getWorkspace()`.

I might missing a path here but actually that should call the ProjectPreferences default constructor what is passing (null,null) and as projectName=null not call getWorkspace ... then we set the workspace and later on it has the correct reference.

Reply:
Hm... it seems my extract into method has not worked as I desired... I'll fix that...

Reply:
I fixed that nwo and using a new commit, so you can see what has changed like in gerrit:

https://github.com/eclipse-platform/eclipse.platform.resources/pull/126/files/0ce3839c1aecc3c883e603308f921aea26f12faf..2f4e5f04dc0ddbfd383bf77884299f7107d56849

Reply:
I see now, thanks. 

So if we want no exception, then: still open:

>  at least we should have a test that verifies the right type returned (in case someone refactors the code, this patch will silently break workspace). Ideally also an error logged in "else" branch.
> For test, something in org.eclipse.core.tests.internal.resources.ProjectPreferencesTest.